src/main/kotlin/com/terraformation/backend/seedbank/db/AccessionStore.kt

```diff
@@ -335,9 +351,11 @@ class AccessionStore(
       throw FacilityNotFoundException(facilityId)
     }
 
+    val speciesId = if (existing.hasDeliveries) existing.speciesId else updated.speciesId
+
     requirePermissions {
       updateAccession(accessionId)
-      updated.speciesId?.let { readSpecies(it) }
+      speciesId?.let { readSpecies(it) }
```

Comment:
Should we return an error code in the case where we don't actually update the species to the one in the payload (i.e. `existing.hasDeliveries` is true)?